crates/bevy_ecs/src/lib.rs

```diff
@@ -1118,6 +1120,28 @@ mod tests {
         query.iter(&world_b);
     }
 
+    #[test]
+    fn query_filters_dont_collide_with_fetches() {
+        let mut world = World::new();
+        world.query_filtered::<&mut i32, Changed<i32>>();
```

Comment:
Should we also add an `Added` variant here too to spot unusual regressions?

Reply:
On the one hand, yes that helps spot unusual regressions. On the other hand, what makes Added more special than With, Without, WithBundle, Or, etc?

Imo we don't need this test to ensure all filter access is implemented correctly (we have other tests for that). We just need to ensure that the filter access doesnt conflict with fetch access (which the test already does).

I think it makes sense to make tests like this small, easy to read, and scoped. So my vote is to leave it as-is (but I concede that there are arguments on both sides).